apps/emqx/src/emqx_router_helper.erl

```diff
@@ -398,15 +399,20 @@ purge_dead_node_trans(Node) ->
     StillKnown = lookup_node_status(Node) =/= notfound,
     case StillKnown andalso node_has_routes(Node) of
         true ->
-            global:trans(
-                {?LOCK(Node), self()},
-                fun() ->
-                    ok = cleanup_routes(Node),
-                    true
-                end,
-                cores(),
-                _Retries = 3
-            );
+            case cores() of
+                Nodes = [_ | _] ->
+                    global:trans(
+                        {?LOCK(Node), self()},
+                        fun() ->
+                            ok = cleanup_routes(Node),
+                            true
+                        end,
+                        Nodes,
+                        _Retries = 3
+                    );
+                undefined ->
+                    error(no_core_nodes)
+            end;
```

Comment:
It's also possible that this replicant has not yet discovered any cores, so `Nodes = []`?

Reply:
Right, but the assumption is that replicant should not end up running this code, purging is cores' job.